### ISSUE TITLE
Publish Java library to maven central

### ISSUE THREAD

**Issue by aceforeverd** [4paradigm/HybridSE#49]
**Is your feature request related to a problem? Please describe.**
Publish Java Library to Maven central

** Requirements **

described in [guide-central-repo-upload](https://maven.apache.org/repository/guide-central-repository-upload.html) and [requirements](https://central.sonatype.org/publish/requirements/). we need
- [x] release: library itself and deps should be release version
- [x] javadoc and source
- [x] pgp sign
- [x] coordinate info: domain ownship
- [x] development info: license, Maintainer，scm
- [x] publish from tag #78

**Comment by aceforeverd** [comment]
need java doc #40

**Comment by aceforeverd** [comment]
for correct coordinates,
+ remove `SNAPSHOT` suffix in version
+ add DNS record for domain valid (require admin help)

**Comment by jingchen2222** [comment]
Please attach to a milestone.

**Comment by aceforeverd** [comment]
# GPG Impl

## brief description
- create two secret in repository `setting>>scrects`
  - `GPG_PRIVATE_KEY`: exported gpg private key, in ASCII mode
  - `GPG_PASSPHRASE`: gpg password
- use `actions/setup-java` to create `settings.xml` for us. It actually access a few environment variables, we will initialize those envs later
- use `crazy-max/ghaction-import-gpg` import the `GPG_PRIVATE_KEY`. You can also write one line shell e.g `gpg --armor --import ${{ secrets.GPG_PRIVATE_KEY }}`, but I'm just to lazy
- run `mvn deploy`

## which gpg secret key ?
Basically, never export the master private key, even github's secrets looks safe. I recommend  generating a gpg subkey and change the password as well. This is also the recommended way from [Debian Development](https://wiki.debian.org/Subkeys).
